src/lib.rs

```diff
@@ -288,24 +289,28 @@ mod tests {
         let frame_id_2 = CANFrame::new(2, &[0u8], false, false).unwrap();
         let frame_id_3 = CANFrame::new(3, &[0u8], false, false).unwrap();
 
-        let (sink, _stream) = socket1.split();
+        let (mut sink, _stream) = socket1.split();
         let (_sink, stream) = socket2.split();
 
-        let take_ids_less_than_3 = stream.take_while(|frame| ok(frame.id() < 3)).for_each(|_| {
-            unsafe { COUNTER += 1 };
-            ok(())
+        let mut take_ids_less_than_3 =
+            stream
+                .try_skip_while(|frame| ok(frame.id() < 3))
+                .map_ok(|frame| {
+                    let id = frame.id().clone();
+                    counter_tx.send(id); // TODO figure out?
```

Comment:
This actually must be awaited.

Reply:
Is the main issue here that trying to `counter_tx.send` consumes the `counter_tx`? Because I haven't played around with the new async/wait syntax or the stabilized futures 0.3 much yet, but I was hoping the pain of needing to keep passing Sink ownership around would be fixed up some.